Python_Scripts/compareStateTrajectories_BioModels_1.py

```diff
@@ -38,66 +30,79 @@ def compStaTraj_BioModels():
-                      'that a newer AMICI version was used!')
-                sys.exit(0)
+            # iterate over the "sedml" models
+            for iMod in range(0, len(list_directory_amici)):
+                iModel = list_directory_amici[iMod]
+                print(iModel)
 
-            for iMod in range(0, len(list_directory_COPASI_bio)):
+                if not os.path.exists(
+                        os.path.join(DIR_MODELS_BIOMODELS, iModel)):
+                    print("Model is not part of Biomodels")
+                    continue
 
-                iModel = list_directory_COPASI_bio[iMod]
-                list_files = sorted(os.listdir(all_biomodels_path + '/' + iModel + '/sbml_models'))
+                list_files = sorted(os.listdir(os.path.join(
+                    DIR_MODELS_SEDML, iModel, 'sbml_models')))
 
+                # iterate over the sbml models
                 for iFile in list_files:
-
-                    # iFile without .xml extension
+                    # iFile without .sbml extension
                     iFile, extension = iFile.split('.', 1)
 
                     # more important model-dependent paths
-                    tsv_path = COPASI_data + '/' + iModel
-                    save_path = base_path + '/biomodels_files/StateTrajectories_BioModels_COPASI_Data/' + iModel
-                    sbml_path = all_biomodels_path + '/' + iModel + '/sbml_models/' + iFile + '.xml'
-
-                    # Open XML file
-                    sbml_model = libsbml.readSBML(sbml_path)
+                    #  path to the original copasi simulation
+                    sbml_path = os.path.join(
+                        DIR_MODELS_SEDML, iModel, 'sbml_models',
+                        iFile + '.sbml')
 
                     # Get whole model
-                    model = all_settings(iModel, iFile, 1)
+                    try:
+                        model = all_settings(iModel, iFile)
+                    except Exception as e:
+                        print(f'Model {iModel} failed ', e)
+                        continue
 
                     ######### COPASI simulatiom
                     # open new .csv file with COPASI simulation trajectories
-                    try:
-                        tsv_file = pd.read_csv(tsv_path + '/Original_COPASI_' + iModel + '_14_14.tsv', sep='\t')
-                    except:
-                        tsv_file = pd.read_csv(tsv_path + '/Original_COPASI_' + iModel + '_12_12.tsv', sep='\t')
+                    if os.path.exists(os.path.join(
+                            DIR_MODELS_TRAJ_REF_BIOMODELS,
+                            f'Original_COPASI_{iModel}_14_14.tsv')):
+                        tsv_file = pd.read_csv(os.path.join(
+                            DIR_MODELS_TRAJ_REF_BIOMODELS,
+                            f'Original_COPASI_{iModel}_14_14.tsv'),
+                            sep='\t')
+                    else:
+                        # TODO Y Why the alternative?
```

Comment:
The tolerances used for creating reference trajectories with Copasi are not homogeneous. Some could be simulated with 14/14, but some failed. Hence, they were simulated with 12/12.
I'm wondering, whether it's meaningful to indicate this in the filename, or rather create one yaml-file or tsv-file, which notes which tolerance combination was used for which model...

Reply:
now the 12_12 or 14_14 is in the "original" file name, but we forget about that detail when copying that file into the "trajectories_reference" folder, so I guess that should be fine as it is.